### ISSUE TITLE
Greater than key with ANSI PCB and ISO-ES layout with OS X

### ISSUE THREAD

**Issue by albertonieto** [pqrs-org/Karabiner-Elements#3010]
Hi,

I use an ANSI keyboard (no extra key next to the left shift) and ISO-ES distribution on OS X.

I want to map the greater key into right_option + period, but I am facing 2 issues:

1. I can't figure out what the greater than key name is
2. I can't make this work (since I don't know the name of the greater than key, I am testing the comma key instead)

```
{
                        "description": "right_option+period to comma",
                        "manipulators": [
                            {
                                "from": [
                                    {
                                        "key_code": "period",
                                        "modifiers": ["right_option"]
                                    }
                                ],
                                "to": {
                                    "key_code": "comma"
                                },
                                "type": "basic"
                            }
                        ]
                    }
```

**Comment by MuhammedZakir** [comment]
Use Karabiner-EventViewer to find out the keycode.

**Comment by albertonieto** [comment]
> 

Well, that's the tricky part. As I was mentioning I am using an ANSI keyboard, so there is no key in between the left shift and the Z keys (there is where the greater than symbol is located on an ISO-ES layout). Therefore, I don't really have a dedicated key for the greater than symbol.

**Comment by MuhammedZakir** [comment]
I see. If you don't have that key, why are you trying to map that key to `right_option+period`? Can you please explain what you're trying to do?

**Comment by albertonieto** [comment]
Sure, I am trying to achieve the following:

I want that when I press right_option+period I get a comma on my text editor (comma is just a placeholder for this test, as I don't know what's the correct key for greater than).

However, with the implemented code shared above, when I press right_control+period (and the event viewer shows exactly that those keys were pressed), nothing happens. No symbol or key gets registered on my text editor.

**Comment by albertonieto** [comment]
I have plugged a Keychron K6 with an ISO PCB (with an extra key between the left shift and the Z key, dedicated to the greater than key). I have opened EventViewer and I can see that the key has the following name: non_us_backslash

With that in mind, and thanks to your advice about the right option, I am planning on changing the combo to right control + period for entering the greater than symbol.

I just need to understand how to implement this on Karabiner. So far, this is the code I am using but it is not working (does nothing):

```
{
                        "description": "right_control+period to >",
                        "manipulators": [
                            {
                                "from": [
                                    {
                                        "key_code": "period",
                                        "modifiers": ["right_control"]
                                    }
                                ],
                                "to": {
                                    "key_code": "non_us_backslash"
                                },
                                "type": "basic"
                            }
                        ]
                    }
```

**Comment by MuhammedZakir** [comment]
> With that in mind, and thanks to your advice about the right option, I am planning on changing the combo to right control + period for entering the greater than symbol.
> 
> I just need to understand how to implement this on Karabiner. So far, this is the code I am using but it is not working (does nothing):

That's plausibly because you're using ANSI layout rather than an ISO layout. I am talking about the layout macOS use, not your keyboard's layout. This depends on the keyboard layout you/macOS use. 

Again: plug your ANSI keyboard, open on-screen Keyboard Viewer and check which key combo results in `>`.

If I understood all this correctly, you are using an ANSI keyboard with an ANSI layout, right? So if you want to enter `>`, then why don't you use `shift+period`? Does that not work? If that works and for some reason you want to use `right_control+period` for `>`, then map `right_control+period` to `shift+period`.
```
                                "to": {
                                    "key_code": "period",
                                    "modifiers": ["shift"]
                                },
```

**Comment by albertonieto** [comment]
Not really, I am using an ANSI 60% PCB with an ISO-ES layout on OS 10.13.6

Now, I am noticing that something is wrong with my Karabiner. Not even this is working:

`{
                        "description": "test",
                        "manipulators": [
                            {
                                "from": [
                                    {
                                        "key_code": "a",
                                        "modifiers": ["shift"]
                                    }
                                ],
                                "to": {
                                    "key_code": "b"
                                },
                                "type": "basic"
                            }
                        ]
                    },`

**Comment by MuhammedZakir** [comment]
> Not really, I am using an ANSI 60% PCB with an ISO-ES layout on OS 10.13.6

Aha! Now I understand what you're trying to do.

> Now, I am noticing that something is wrong with my Karabiner. Not even this is working:
> 
> `{ "description": "test", "manipulators": [ { "from": [ { "key_code": "a", "modifiers": ["shift"] } ], "to": { "key_code": "b" }, "type": "basic" } ] },`

Have you enabled input modification for your device in KE pereferences? It should be in `Devices` tab IIRC.

**Comment by albertonieto** [comment]
Not sure, this is what I have there [https://app.box.com/s/rvm8nqujzcfpqfhlfdjlypjj7zgfvvl3](screenshot)

**Comment by MuhammedZakir** [comment]
> Not sure, this is what I have there https://app.box.com/s/rvm8nqujzcfpqfhlfdjlypjj7zgfvvl3

If you are only using one keyboard, but KE shows 2 devices - 1. keyboard icon and 2. keyboard+mouse icons, enable both. I think you need to only enable `#2`, but I don't remember. You can try enabling one and disabling the other, and see if modification works (you can test them using simple modification). If it doesn't, do the same for the other. Or, just enable both of them. I don't think it will cause any problems.

**Comment by albertonieto** [comment]
I think this may be one of those irresoluble things.

Since I am using a 60% form factor (ANSI layout with 1u right shift and arrows), I do not have the F1-F12 row (also, the key before 1 is ESC). And since this is an ANSI keyboard there is no key between the left shift and the Z key.

On top of that, the ISO-ES input source on my OS X system is positioning the greater than key in the same position where the ESC key is located. How am I supposed to use the greater than key if ESC is the key there? Well, that is what the FN key is for.

With this 60% keyboard on OS X with an ISO-ES input source, by default, I have to use FN+ESC to output < or FN+SHIFT+ESC to output >

Why do I think this is irresoluble? Well, for 2 main reasons:

1. Because Karabiner is unable to register the FN key (I can't code something like `{ "from": [ { "key_code": "period", "modifiers": ["right_control"] } ], "to": { "key_code": "ESC", "modifiers": ["shift", "fn" }`).
2. Because OS X is unable to position the <> key somewhere else.
